sdks/java/core/src/main/java/org/apache/beam/sdk/util/GcsUtil.java

```diff
@@ -349,16 +350,45 @@ public boolean bucketExists(GcsPath path) throws IOException {
   }
 
   /**
+   * Returns the project number of the project which owns this bucket.
+   * If the bucket exists, it must be accessible otherwise the permissions
+   * exception will be propagated.
+   */
+  public long bucketOwner(GcsPath path) throws IOException {
+    return getBucket(
+        path,
+        BACKOFF_FACTORY.backoff(),
+        Sleeper.DEFAULT).getProjectNumber().longValue();
+  }
+
+  /**
+   * Creates a bucket for the provided project or propagates an error.
+   */
+  public void createBucket(GcsPath path, long projectNumber) throws IOException {
+    createBucket(
+        path,
+        projectNumber,
+        BACKOFF_FACTORY.backoff(),
+        Sleeper.DEFAULT);
+  }
+
+  /**
    * Returns whether the GCS bucket exists. This will return false if the bucket
    * is inaccessible due to permissions.
    */
   @VisibleForTesting
   boolean bucketExists(GcsPath path, BackOff backoff, Sleeper sleeper) throws IOException {
```

Comment:
Given the way we actually use this, it seems like it would be better named `bucketAccessible`